scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -1246,42 +1250,58 @@ class FormatOps(
 
   def getLambdaAtSingleArgCallSite(
       ft: FormatToken
-  )(implicit style: ScalafmtConfig): Option[Term.Function] =
+  )(implicit style: ScalafmtConfig): Option[Tree] = {
+    def infixArg(fun: Tree) = fun.parent.exists({
+      case Term.ApplyInfix(_, _, _, List(`fun`)) => true
+      case _ => false
+    })
     ft.meta.leftOwner match {
       case Term.Apply(_, List(fun: Term.Function)) => Some(fun)
-      case fun: Term.Function if fun.parent.exists({
-            case Term.ApplyInfix(_, _, _, List(`fun`)) => true
-            case _ => false
-          }) =>
-        Some(fun)
+      case Term.Apply(_, List(fun: Term.ContextFunction)) => Some(fun)
+      case fun: Term.Function if infixArg(fun) => Some(fun)
+      case fun: Term.ContextFunction if infixArg(fun) => Some(fun)
       case t: Init =>
-        findArgsFor(ft.left, t.argss).collect { case List(f: Term.Function) =>
-          f
+        val args = findArgsFor(ft.left, t.argss)
```

Comment:
isn't this unnecessarily complex? why not just add another case to the original collect?